cmd/eval-dev-quality/cmd/evaluate_test.go

```diff
@@ -670,4 +672,139 @@ func TestEvaluateExecute(t *testing.T) {
 			filepath.Join("empty-response", "golang", "golang", "plain.log"): nil,
 		},
 	})
+
+	t.Run("Retries", func(t *testing.T) {
+		{
+			queryMock := providertesting.NewMockQuery(t)
+			llmModel := llm.NewModel(queryMock, "testing-provider/testing-model")
+			providerMock := providertesting.NewMockProviderNamedWithModels(t, "testing-provider", []model.Model{llmModel})
+			validate(t, &testCase{
+				Name: "Single try fails",
+
+				Before: func(t *testing.T, logger *log.Logger, resultPath string) {
+					queryMock.On("Query", mock.Anything, "testing-provider/testing-model", mock.Anything).Return("", errors.New("empty response from model"))
+
+					provider.Register(
+						&struct {
+							provider.Provider
+							provider.Query
+						}{
+							Provider: providerMock,
+							Query:    queryMock,
+						},
+					)
+				},
+				After: func(t *testing.T, logger *log.Logger, resultPath string) {
+					delete(provider.Providers, "testing-provider")
+
+					queryMock.AssertNumberOfCalls(t, "Query", 1)
+				},
+				Arguments: []string{
+					"--language", "golang",
+					"--model", "testing-provider/testing-model",
+					"--retries", "1",
+				},
+				ExpectedResultFiles: map[string]func(t *testing.T, filePath string, data string){
+					"categories.svg":    nil,
+					"evaluation.csv":    nil,
+					"evaluation.log":    nil,
+					"golang-summed.csv": nil,
+					"models-summed.csv": nil,
+					"README.md":         nil,
+					filepath.Join("testing-provider_testing-model", "golang", "golang", "plain.log"): func(t *testing.T, filePath, data string) {
+						assert.Contains(t, data, "empty response from model")
+					},
+				},
+			})
+		}
+		{
+			queryMock := providertesting.NewMockQuery(t)
+			llmModel := llm.NewModel(queryMock, "testing-provider/testing-model")
+			providerMock := providertesting.NewMockProviderNamedWithModels(t, "testing-provider", []model.Model{llmModel})
+			validate(t, &testCase{
+				Name: "Work afterwards",
+
+				Before: func(t *testing.T, logger *log.Logger, resultPath string) {
+					queryMock.On("Query", mock.Anything, "testing-provider/testing-model", mock.Anything).Return("", errors.New("empty response from model")).Once()
+					queryMock.On("Query", mock.Anything, "testing-provider/testing-model", mock.Anything).Return("model-response", nil).Once().After(10 * time.Millisecond)
```

Comment:
Let's also add a comment here on why we have the delay: `Simulate a model response delay because our internal safety measures trigger when a query is done in 0 milliseconds.` (same for the case below)